src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -11,20 +14,86 @@ namespace trajectory
 {
 
 using ostk::physics::time::Duration;
+using EarthGravitationalModel = ostk::physics::environment::gravitational::Earth;
 
 using ostk::astro::trajectory::Propagator;
+using ostk::astro::trajectory::state::CoordinatesSubset;
 
 Segment::Solution::Solution(
     const String& aName,
     const Array<Shared<Dynamics>>& aDynamicsArray,
     const Array<State>& aStates,
-    const bool& aConditionIsSatisfied
+    const bool& aConditionIsSatisfied,
+    const Segment::Type& aSegmentType
 )
     : name(aName),
       dynamics(aDynamicsArray),
       states(aStates),
-      conditionIsSatisfied(aConditionIsSatisfied)
+      conditionIsSatisfied(aConditionIsSatisfied),
+      segmentType(aSegmentType)
+{
+}
+
+const Instant& Segment::Solution::accessStartInstant() const
+{
+    if (states.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("States are empty.");
```

Comment:
```suggestion
        throw ostk::core::error::RuntimeError("No solution available.");
```

Is this condition relevant when the Segment hasn't been solved yet?

If so, maybe an error that says something like "Segment not yet solved." would be a little more helpful.

Also, for this one specifically, is it possible for the startInstant to be available when there aren't states or nah?

Reply:
Yeah good point. Technically if we run solve it should atleast have 1 state even if the condition is met immediately.